server/db/index.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const util = require('util');
+const path = require('path');
+const fs = require('fs');
+
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
+
+const moviesPath = path.resolve('server/db/movies.json');
+
+// Reads and returns contents of movies.json
+async function readMovies() {
+  return readFile(moviesPath)
+    .then((contents) => {
+      return JSON.parse(contents);
+    });
+}
+
+// Replaces the contents of movies.json
+async function writeMovies(movies) {
+  const json = JSON.stringify(movies, null, 2);
+  return writeFile(moviesPath, json);
+}
+
+// Gets a list of each unique movie
+async function getAllTitles() {
+  return readMovies()
+    .then((movies) => {
+      const names = [];
+      movies.forEach((movie) => {
+        if (!names.includes(movie.name)) {
```

Comment:
Instead of storing duplicate titles why not just ensure you store no duplicate titles or allow the updating of them?